code/drasil-gool/lib/Drasil/Shared/RendererClassesCommon.hs

```diff
@@ -124,12 +130,16 @@ class OpElim r where
   uOpPrec :: r (UnaryOp r) -> Int
   bOpPrec :: r (BinaryOp r) -> Int
 
+-- These assume we represent Scope with ScopeData
 class ScopeElim r where
   scopeData :: r (Scope r) -> ScopeData
-  
+
+-- What exactly is this, since it adds a bunch of extra required data?
 class RenderVariable r where
   varFromData :: Binding -> String -> VSType r -> Doc -> SVariable r
-    
+
+-- This assumes variables are represented in a certain way, and
+-- its existence seems a bit sketchy
```

Comment:
It is likely used for doing a bit of introspection. Might not be needed if the information is available by other means.